seth/tts-CalculatorAssignment/JsKoo.js

```diff
@@ -0,0 +1,114 @@
+var onebtn = document.getElementById('calc-one');
+var onebtn = document.getElementById('calc-two');
+var onebtn = document.getElementById('calc-three');
+var onebtn = document.getElementById('calc-four');
+var onebtn = document.getElementById('calc-five');
+var onebtn = document.getElementById('calc-six');
+var onebtn = document.getElementById('calc-seven');
+var onebtn = document.getElementById('calc-eight');
+var onebtn = document.getElementById('calc-nine');
+var onebtn = document.getElementById('calc-zero');
+
+var decimalbtn = document.getElementById('decimal');
+var clearbtn = document.getElementById('clear');
+var backpacebtn = document.getElementById('backspace');
+var displayValElement = document.getElementById('Calculator-display');
+
+var displayVal = '0';
+var pendingVal;
+var evalStringArray = [];     /* problem here */
+
+var calcNumnbtns = document.getElementsByClassName('calc-btn-num');
+var calOperationBtns = document.getElementsByClassName('calc-btn-operator');
+
+var updateDisplayVal = (clickObj) => {
+  var btnText = clickObj.target.innerText;
+
+if(displayVal === '0')
+  displayVal = '';
+
+displayVal += btnText;
+displayValElement.innerText = displayVal;
+}
+
+var performOperation = (clickObj) => {
+  var operator = clickObj.target.innerText;
+
+    switch (operator) {
+      case '+':
+        pendingVal = displayVal;
+        display ='0';
+        displayValElement.innerText = displayVal;
+        evalStringArray.push(pendingVal);
```

Comment:
There's a lot of indirection going on in here. You have variables that feel a bit unnecessary, setting them, then pushing them onto this array. It would be a bit simpler just to `evalStringArray.push(displayVal)` rather than assign pendingVal to display val then push it. Unless I'm missing something =)